### ISSUE TITLE
Returning `null` from non-nullable local function should warn

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#26639]
```C#
class Program
{
    void M2()
    {
        M();
        
        string M()
        {
            return null; // expect a warning, just like in regular methods
        }
    }
}
```

Found in nullable dogfood.

Note: there are some IDE tests (in `CSharpDeclareAsNullableCodeFixTests`) for this bug, which should be unskipped.

**Comment by cston** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/28103.
